java/src/jmri/ThrottleListener.java

```diff
@@ -38,67 +63,113 @@
      * @param reason  The reason why the throttle request failed.
      */
     public void notifyFailedThrottleRequest(LocoAddress address, String reason);
-
+    
     /**
-     * Get notification that a throttle request requires is in use by another
-     * device, and a "steal" may be required.
+     * Get notification that a throttle request is in use by another
+     * device, and a "steal", "share", or "steal/share" decision may be required.
      *
-     * @param address LocoAddress of the throttle that needs to be stolen.
+     * @param address The LocoAddress that needs the decision.
+     * @param question The question being asked, steal / cancel, share / cancel, steal / share / cancel
      */
-    public void notifyStealThrottleRequired(LocoAddress address);
```

Comment:
Because this is a public method that is part of the JMRI API and it has been in production releases, you can’t just remove this.  You need to deprecate it and it needs to stick around for at least 2 release cycles.

You can put in a default method that calls the replacement with appropriate options.

Reply:
These should all now be marked up as appropriate.